features/support/commits_finder.rb

```diff
@@ -0,0 +1,49 @@
+# Encapsulates finding Git commits in flexible ways
+#
+# After initializing with the desired attributes for the commit list,
+# run as many #find_* methods as you want to build up your commit list.
+# When done, call #to_a to get the results as a Cucumber compatible array.
+class CommitsFinder
```

Comment:
Btw this finder needs internal mutable state, because complex finder operations like "find all commits in a repo" can be comprised of several other find operations ("find all commits in a branch"). 

We could make the `find_commits_in_current_repo` method completely stateless through a map-reduce architecture: first determine all commits in all branches through a map of `find_commits_in_branch` over all branch names in the repo, then reduce all those results into a single CommitsList. This would be more complex and more wasteful though, so I'm hesitant to do it given that the existing implementation works well and is simpler.

Since we use a new CommitsFinder instance for each search operation, this mutable state is localized (not shared), and doesn't change after the search operation is over. Also, it is encapsulated in its own class. So I think this pattern is appropriate here.